### ISSUE TITLE
Can not restart the workspace from the local devfile

### ISSUE THREAD

**Issue by jorgeadrian93** [eclipse-che/che#22768]
### Describe the bug

After making a small change to my devfile (like add a new env var) to test the option: `Dev Spaces: Restart Workspace from Local Devfile`. The workspace does not restart and displays error messages such as: `aborted`, `http request failed` (see attached images) 

I have already tested this with multiple projects and basic devfiles but with the same result. 

### Che version

7.77

### Steps to reproduce

devfile content when starting the workspace: 

```
schemaVersion: 2.2.0
metadata:
  name: my-app
attributes:
  controller.devfile.io/storage-type: per-workspace
components:
- name: tools
  container:
    image: quay.io/devfile/universal-developer-image:ubi8-latest
    mountSources: true
    sourceMapping: /projects
    volumeMounts:
    - name: home
      path: /home/user/
    memoryLimit: 6G
    memoryRequest: 2G
- volume:
    size: 10Gi
  name: home

commands:
  - id: package
    exec:
      label: "Package"
      component: tools
      workingDir: ${PROJECT_SOURCE}
      commandLine: "mvn clean package"
      group:
        kind: build
        isDefault: true
```

### Expected behavior

Add an environment variable and restart workspace from local devfile

```
schemaVersion: 2.2.0
metadata:
  name: my-app
attributes:
  controller.devfile.io/storage-type: per-workspace
components:
- name: tools
  container:
    image: quay.io/devfile/universal-developer-image:ubi8-latest
    env:
      - name: TEST_RESTART
        value: 'Restart from Local DevFile'
    mountSources: true
    sourceMapping: /projects
    volumeMounts:
    - name: home
      path: /home/user/
    memoryLimit: 6G
    memoryRequest: 2G
- volume:
    size: 10Gi
  name: home

commands:
  - id: package
    exec:
      label: "Package"
      component: tools
      workingDir: ${PROJECT_SOURCE}
      commandLine: "mvn clean package"
      group:
        kind: build
        isDefault: true
```

### Runtime

OpenShift

### Screenshots

![image](https://github.com/eclipse/che/assets/19197341/219bcb08-6a77-41c1-afda-abcb03099422)

![image](https://github.com/eclipse/che/assets/19197341/98d09508-6a6c-4c5a-bdbe-8ce612477504)

![image](https://github.com/eclipse/che/assets/19197341/3ab106e6-9f93-4f5d-b566-d7a68425f358)

### Installation method

OperatorHub

### Environment

Linux

### Additional context

I found these related issues: 
https://github.com/eclipse/che/issues/22453
https://github.com/eclipse/che/issues/22722
https://github.com/eclipse/che/issues/22432

**Comment by AObuchow** [comment]
@jorgeadrian93 Thank you for reporting this.

I need to do some testing to figure out why you're receiving this error and will report back here.

**Comment by ibuziuk** [comment]
@jorgeadrian93 according to the screenshots Dev Spaces 3.10 version is used, and restart from local devfile is supposed to work on this version - you can try it out on workspaces.openshift.com.

Indeed, if there is a proxy configured on a cluster, there is currently a known issue, that is supposed to be fixed in the next version of Dev Spaces - https://issues.redhat.com/browse/CRW-5421


The workaround is to update CR accordingly:

```
spec:
  components:
    cheServer:
      proxy:
        nonProxyHosts:
          - "172.30.0.1" 
 ```

basically,  `KUBERNETES_SERVICE_HOST` needed to be added to the no-proxy list by default. We did a similar exercise in DWO some time ago, and now need to do the same for the operator - https://github.com/devfile/devworkspace-operator/pull/840

**Comment by jorgeadrian93** [comment]
Awesome! With the addition of the no-proxy rule as described by @ibuziuk , it is now working 🚀  Thanks a TON y'all.
